### PR TITLE
[Core] Enable TPU Autoscaling with Kuberay

### PR DESCRIPTION
**Pull request by ryanaoleary**
## Why are these changes needed?

In order to enable TPU Pod autoscaling, the Kuberay NodeProvider needs to be aware of the `TPU-{...}-head` resource (similar to the changes made to support TPU pod autoscaling in GCP here: https://github.com/ray-project/ray/pull/40463) to enable the autoscaler to make scaling decisions.

This PR adds functionality to determine the TPU accelerator type based on GKE TPU nodeSelectors and adds a `TPU-{accelerator_type}-head` resource to the autoscaling config. The number of TPU heads is determined by the replicas of the TPU worker group, since each worker group is treated as a TPU podslice. This PR also adds a TPU worker group to `test_autoscaling_config`.
## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests: using [detached actors](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/configuring-autoscaling.html#step-5-trigger-raycluster-scale-up-by-creating-detached-actors) requesting TPUs 
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ryanaoleary** [comment]
PR for https://github.com/ray-project/ray/pull/45105 that now skips the release tests on Windows that were failing

**Comment by jjyao** [comment]
Do we know the cause of the failure? By default we should fix instead of skip.

**Comment by ryanaoleary** [comment]
> Do we know the cause of the failure? By default we should fix instead of skip.

The logs Kai-Hsun sent me just show: 
```
No such file or directory: 'C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\Bazel.runfiles_y2gns_5d\\runfiles\\com_github_ray_project_ray\\python\\ray\\autoscaler\\kuberay\\ray-cluster.complete.yaml'
```
I'm not sure why this is the case, but all the existing tests in [test_autoscaling_config](https://github.com/ray-project/ray/blob/036efa66697f76ffdeea3a56b58ea705b5eac8e6/python/ray/tests/kuberay/test_autoscaling_config.py) specify `@pytest.mark.skipif(platform.system() == "Windows", reason="Not relevant.")` so I added that to the test I created.

**Review by jjyao — APPROVED** [review]

File: python/ray/tests/kuberay/test_autoscaling_config.py [hunk 1]
```diff
@@ -336,6 +422,61 @@ def _fetch_ray_cr_from_k8s(self) -> Dict[str, Any]:
...
 
+TPU_PARAM_ARGS = ",".join(
+    [
+        "ray_cr_in",
+        "expected_num_tpus",
+    ]
+)
+TPU_TEST_DATA = (
+    []
+    if platform.system() == "Windows"
```
Comment (jjyao) [inline python/ray/tests/kuberay/test_autoscaling_config.py:433]: No need to have the if else here since we already have `skipif` down below.

Comment (ryanaoleary) [reply python/ray/tests/kuberay/test_autoscaling_config.py]: Done in 9d186c5

**Pull request merged**